adb_graphics/datahandler/gribdata.py

```diff
@@ -481,128 +286,185 @@ class fieldData(UPPData):
-            if callable(ret):
-                return ret()
-            return ret
-        except AttributeError:
-            return color_spec
+        except AttributeError as e:
+            msg = f"There is no color definition named {color_spec}"
+            raise AttributeError(msg) from e
+        if callable(ret):
+            return np.asarray(ret())
+        return np.asarray(ret)
 
     @property
     def corners(self) -> list:
+        """
 
-        '''
-        Returns lat and lon of lower left (ll) and upper right(ur) corners:
+        Returns lat and lon of lower left (ll) and upper right (ur) corners.
+
+        Order:
                ll_lat, ur_lat, ll_lon, ur_lon
-        '''
+        """
 
         lat, lon = self.latlons()
-        if self.model in ['global', 'hfip', 'obs']:
-            ret = [lat[-1], lat[0], lon[0], lon[-1]]
-        elif self.model == 'global_mpas':
-            ret = [lat[0], lat[-1], lon[0], lon[-1]]
-        else:
-            ret = [lat[0, 0], lat[-1, -1], lon[0, 0], lon[-1, -1]]
+        if len(lat.shape) == 2:
+            return [
+                np.round(x, decimals=6) for x in [lat[0, 0], lat[-1, -1], lon[0, 0], lon[-1, -1]]
+            ]
+        return [np.round(x, decimals=6) for x in [lat[0], lat[-1], lon[0], lon[-1]]]
 
-        return ret
+    @property
+    def data(self) -> DataArray:
+        """
+        Sets the data property on the object for use when we need to update
+        the values associated with a given object -- helpful for differences.
+        """
+        if not hasattr(self, "_data"):
+            self._data = self.values()
+        return self._data
+
+    @data.setter
+    def data(self, value: DataArray):
+        self._data = value
+
+    def field_column_max(self, values: DataArray, **_kwargs):
+        """Returns the column max of the values."""
+
+        return values.max(dim=self.vertical_coord)
+
+    def field_diff(self, values: DataArray, variable2: str, level2: str, **kwargs):
+        """Subtracts the values from variable2 from self.field."""
+
+        value2 = self.values(
+            name=variable2, level=level2, do_transform=kwargs.get("do_transform", True)
+        )
+        diff = values - value2
+        value2.close()
+
+        return diff
 
-    def fire_weather_index(self, values, **kwargs):
+    def field_mean(
+        self,
+        values: DataArray,
+        levels: list,
+        **kwargs,
+    ):
+        """Returns the mean of the values over the vertical dimension."""
 
-        # pylint: disable=unused-argument
+        levels = kwargs["global_levels"] if "global" in self.model else levels
+        levs = [int(x[:-2]) for x in levels]
+        return values.sel(isobaricInhPa=levs).mean("isobaricInhPa")
 
-        '''
-        Generates a field of Fire Weather Index
+    def field_sum(self, values: DataArray, variable2: str, level2: str, **kwargs):
+        """Return the sum of the values."""
```

Comment:
```suggestion
        """Returns the sum of the values."""
```
For consistency with other docstrings.